grudge/op.py

```diff
@@ -478,13 +478,36 @@ def _cross_rank_trace_pairs_scalar_field(dcoll, vec, tag=None):
 
 
 def cross_rank_trace_pairs(dcoll, vec, tag=None):
+    """Get a list of *vec* trace pairs for each partition boundary.
+
+    For each partition boundary, the field data values in *vec* are
+    communicated to/from the neighboring partition. Presumably, this
+    communication is MPI (but strictly speaking, may not be, and this
+    routine is agnostic to the underlying communication, see e.g.
+    _cross_rank_trace_pairs_scalar_field).
+
+    For each face on each partition boundary, a :class:`TracePair` is
+    created with the locally, and remote owned partition boundary face
+    data as the `internal`, and `external` components, respectively.
+    Each of the TracePair components are structured like *vec*.
+
+    The input field data *vec* may be a single DOFArray, or an object
+    array of DOFArrays. Each of *vec* components are independently
+    communicated by calling this routine. Upon entry, *vec* is
+    serialized (if needed), each component is communicated, then
+    (if needed) the components are de-serialized back to the original
+    structure of *vec*, before returned as TracePairs for each partition
+    boundary.
```

Comment:
```suggestion
    array of ``DOFArray``\ s of arbitrary shape.
```
(I don't think it matters to users exactly how the documented effect is achieved, so I don't think we should specify it, especially when the implementation, as it is here, is a tad *cough* suboptimal.)